torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -249,28 +248,36 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(replacement_proxy.node)
                     graph.erase_node(node)
 
-    # DCE symbols (which are guaranteed to be pure) only
-    graph.eliminate_dead_code()
-
-    # Now do one more pass that specializes all symfloats we didn't manage
-    # to tensorify away.
     nodes = list(graph.nodes)
-    for i, node in enumerate(nodes[:-1]):
+    for i, node in enumerate(nodes[-2::-1]):
```

Comment:
It's probably clearer to do reversed here, and also axe the slicing in favor of just testing if a node.op is output and not DCE'ing in that case